src/auth/PrivateRoute.tsx

```diff
@@ -95,12 +97,14 @@ const PrivateRouteWrapper: React.FC<{
         authCheckCompleted.current = false;
         return <div key="children">{props.children}</div>;
       } else {
-        return <Redirect to="/login/" />;
+        return <Error404 />;
       }
     } else {
       return <>
-        {props.fallback ?? <h1>Loading...</h1>}
-        <div key="children" className="d-none">{props.children}</div>
+        {props.fallback ?? <Spinner animation="border" role="status">
+          <span className="sr-only">Loading...</span>
+        </Spinner>}
+        <div key="children" className="d-none" aria-hidden="true">{props.children}</div>
```

Comment:
This is maybe an admission that I didn't notice this when we initially reviewed, but why are we hiding the children here instead of just not rendering them?

Reply:
Performance --- we can do a bit of mounting/rendering/API calls while the spinner is visible.